executor/executor.cc

```diff
@@ -732,30 +752,36 @@ void cover_enable(thread_t* th)
 	if (!flag_cover)
 		return;
 	debug("#%d: enabling /sys/kernel/debug/kcov\n", th->id);
-	if (ioctl(th->cover_fd, KCOV_ENABLE, 0)) {
-		// This should be fatal,
-		// but in practice ioctl fails with assorted errors (9, 14, 25),
-		// so we use exitf.
-		exitf("cover enable write failed");
+	if (flag_collect_comps) {
+		if (ioctl(th->cover_fd, KCOV_ENABLE, KCOV_MODE_TRACE_CMP)) {
+			exitf("cover enable write trace cmp failed");
+		}
+	} else {
+		if (ioctl(th->cover_fd, KCOV_ENABLE, KCOV_MODE_TRACE_PC)) {
+			// This should be fatal,
+			// but in practice ioctl fails with assorted errors (9, 14, 25),
+			// so we use exitf.
+			exitf("cover enable write trace pc failed");
+		}
 	}
-	debug("#%d: enabled /sys/kernel/debug/kcov\n", th->id);
+	debug("#%d: enabled /sys/kernel/debug/kcov with comps=%u\n", th->id,
+	      flag_collect_comps);
```

Comment:
We already printed all flags at the beginning of execution. No need to print them second time.